letsencrypt/models/letsencrypt.py

```diff
@@ -2,167 +2,255 @@
+        # Generate account key
+        account_key_file = self._generate_key('account.key')
+        account_key = jose.JWKRSA.load(open(account_key_file).read())
+        # Generate domain key
+        domain_key_file = self._generate_key(domain)
+        client = self._create_client(account_key)
+        new_reg = NewRegistration(
+            key=account_key.public_key(),
+            terms_of_service_agreed=True)
+        client.new_account(new_reg)
+        csr = self._make_csr(account_key, domain_key_file, domain)
+        authzr = client.new_order(csr)
+        # write the file with the challenge
+        challenge_dir = _get_challenge_dir()
+        if not isdir(challenge_dir):
+            makedirs(challenge_dir)
+        auth_responded = False
+        for authorizations in authzr.authorizations:
+            for challenge in sorted(
+                    authorizations.body.challenges, key=lambda x: x == 'http'):
+                if challenge.chall.typ == TYPE_CHALLENGE_HTTP:
+                    self._respond_challenge_http(challenge, account_key)
+                    client.answer_challenge(
+                        challenge,
+                        challenges.HTTP01Response())
+                    auth_responded = True
+                    break
+                elif challenge.chall.typ == TYPE_CHALLENGE_DNS:
+                    self._respond_challenge_dns(challenge, domain)
+                    client.answer_challenge(
+                        challenge, challenges.DNSResponse())
+                    auth_responded = True
+                    break
+        if not auth_responded:
+            raise exceptions.ValidationError(
+                _('Could not respond to letsencrypt challenges.'))
+        # let them know we are done and they should check
+        deadline = datetime.now() + timedelta(minutes=5)
+        order_resource = client.poll_and_finalize(authzr, deadline)
+        with open(join(_get_data_dir(), '%s.crt' % domain), 'w') as crt:
+            crt.write(order_resource.fullchain_pem)
+            _logger.info('SUCCESS: Certificate saved :%s', crt.name)
+            ir_config_parameter = self.env['ir.config_parameter']
+            reload_cmd = ir_config_parameter.get_param(
+                'letsencrypt_reload_command', False)
+            if reload_cmd:
+                self._call_cmdline(['sh', '-c', reload_cmd])
+            else:
+                _logger.warning("No reload command defined.")
+
+    def _create_client(self, account_key):
+        net = client.ClientNetwork(account_key)
+        if config['test_enable']:
+            directory_url = V2_STAGING_DIRECTORY_URL
+        else:
+            directory_url = V2_DIRECTORY_URL
+        directory_json = requests.get(directory_url).json()
+        return client.ClientV2(directory_json, net)
+
+    def _cascade_domains(self, altnames):
+        if len(altnames) > 1:
+            for altname in filter(lambda x: '*.' in x, altnames):
```

Comment:
@hbrunn What do you mean anywhere? It can only be next to the dot. The wildcard is asterisk plus dot. 

> A wildcard domain name consists
>    of a single asterisk character followed by a single full stop
>    character ("*.")
https://www.ietf.org/id/draft-ietf-acme-acme-18.txt

Reply:
my rfc is https://tools.ietf.org/html/rfc6125#section-6.4.3 - but as it seems acme doesn't support this, do it the acme way